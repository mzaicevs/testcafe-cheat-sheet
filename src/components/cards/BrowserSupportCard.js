import React from "react";
import TableCard from "../../base_components/TableCard";
import { CodeHighlight } from "../../base_components/CodeHighlight";
import { browserSupport } from "../../snippets";

const BrowserSupportCard = () => (
  <TableCard
    cardTitle={browserSupport.title}
    headers={["Browser", "Alias"]}
    rows={browserSupport.browsers.map(browser => [
      browser.text,
      browser.aliases.map(alias => (
        <CodeHighlight key={alias} language="plaintext">
          {alias}
        </CodeHighlight>
      ))
    ])}
  />
);
export default BrowserSupportCard;
