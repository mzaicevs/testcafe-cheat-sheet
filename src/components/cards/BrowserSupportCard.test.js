import React from "react";
import { shallow } from "enzyme";
import BrowserSupportCard from "./BrowserSupportCard";

describe("BrowserSupportCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<BrowserSupportCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
