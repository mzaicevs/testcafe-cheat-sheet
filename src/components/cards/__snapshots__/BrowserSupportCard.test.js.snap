// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserSupportCard matches snapshot 1`] = `
<TableCard
  cardTitle="Browser Support"
  headers={
    Array [
      "Browser",
      "Alias",
    ]
  }
  rows={
    Array [
      Array [
        "Google Chrome: Stable, Beta, Dev and Canary",
        Array [
          <CodeHighlight
            language="plaintext"
          >
            chromium
          </CodeHighlight>,
          <CodeHighlight
            language="plaintext"
          >
            chrome
          </CodeHighlight>,
          <CodeHighlight
            language="plaintext"
          >
            chrome-canary
          </CodeHighlight>,
          <CodeHighlight
            language="plaintext"
          >
            chrome:headless
          </CodeHighlight>,
        ],
      ],
      Array [
        "Internet Explorer (11+)",
        Array [
          <CodeHighlight
            language="plaintext"
          >
            ie
          </CodeHighlight>,
        ],
      ],
      Array [
        "Microsoft Edge",
        Array [
          <CodeHighlight
            language="plaintext"
          >
            edge
          </CodeHighlight>,
        ],
      ],
      Array [
        "Mozilla Firefox",
        Array [
          <CodeHighlight
            language="plaintext"
          >
            firefox
          </CodeHighlight>,
          <CodeHighlight
            language="plaintext"
          >
            firefox:headless
          </CodeHighlight>,
        ],
      ],
      Array [
        "Safari",
        Array [
          <CodeHighlight
            language="plaintext"
          >
            safari
          </CodeHighlight>,
        ],
      ],
      Array [
        "Opera",
        Array [
          <CodeHighlight
            language="plaintext"
          >
            opera
          </CodeHighlight>,
        ],
      ],
    ]
  }
/>
`;
